tests/unittest/test_argument_parsing.py

```diff
@@ -15,197 +14,211 @@
-            ["delete", "--username", "spark-2", "--namespace", "ns2"]
-        )
-
-        self.assertEqual(args.action, "delete")
-        self.assertEqual(args.username, "spark-2")
-        self.assertEqual(args.namespace, "ns2")
-
-    def test_delete_conf(self):
-        args = self.parser.parse_args(
-            ["clear-config", "--username", "spark-2", "--namespace", "ns2"]
-        )
-
-        self.assertEqual(args.action, "clear-config")
-        self.assertEqual(args.username, "spark-2")
-        self.assertEqual(args.namespace, "ns2")
-
-    def test_get_conf(self):
-        args = self.parser.parse_args(["get-config"])
-
-        self.assertEqual(args.action, "get-config")
-        self.assertEqual(args.username, "spark")
-        self.assertEqual(args.namespace, "default")
-
-    def test_add_conf(self):
-        args = self.parser.parse_args(
-            [
-                "add-config",
-                "--username",
-                "spark-3",
-                "--conf",
-                "mykey=myvalue",
-                "--properties-file",
-                "my-file",
-            ]
-        )
-
-        self.assertEqual(args.action, "add-config")
-        self.assertEqual(len(args.conf), 1)
-        self.assertEqual(args.conf[0], "mykey=myvalue")
-        self.assertEqual(args.username, "spark-3")
-        self.assertEqual(args.namespace, "default")
-        self.assertEqual(args.properties_file, "my-file")
-
-    def test_list(self):
-        args = self.parser.parse_args(["list", "--kubeconfig", "my-kube-config"])
-
-        self.assertEqual(args.action, "list")
-        self.assertEqual(args.kubeconfig, "my-kube-config")
-
-        args = self.parser.parse_args(["list"])
-
-        self.assertEqual(args.action, "list")
-        self.assertEqual(args.kubeconfig, None)
-
-    def test_primary(self):
-        args = self.parser.parse_args(["get-primary", "--master", "my-k8s-api-server"])
-
-        self.assertEqual(args.action, "get-primary")
-        self.assertEqual(args.master, "my-k8s-api-server")
-
-        args = self.parser.parse_args(["get-primary"])
-
-        self.assertEqual(args.action, "get-primary")
-        self.assertEqual(args.master, None)
-
-
-if __name__ == "__main__":
-    logging.basicConfig(format="%(asctime)s %(levelname)s %(message)s", level="DEBUG")
-    unittest.main()
+@pytest.fixture
+def shell_parser():
+    return parse_arguments_with(
+        [
+            add_logging_arguments,
+            k8s_parser,
+            spark_user_parser,
+            add_config_arguments,
+        ],
+        argparse.ArgumentParser(exit_on_error=False),
+    )
+
+
+@pytest.fixture
+def submit_parser():
+    return parse_arguments_with(
+        [
+            add_logging_arguments,
+            k8s_parser,
+            spark_user_parser,
+            add_deploy_arguments,
+            add_config_arguments,
+        ],
+        argparse.ArgumentParser(exit_on_error=False),
+    )
```

Comment:
**nitpick:** optional, personally prefer to have these in some `conftest.py`, leaving test files to tests wherever possible.

Reply:
Yes, generally me too. Normally I'm keeping fixtures in the test module, that are kinda very specific to the test module -- in attempt to 

 1. keep `conftest.py` from "pollution"
 2. make it more readable for tests, to be able to re-visit their less intuitive fixtures on the spot.

However, here I need to confess all I did was "brainlessly" converting the original module to a different syntax.
These two could be debateable (as for now they are a bit tied to the only argument parsing module we have)... 
But after all may be also good for re-use, so I put them in a unittest-level `conftest.py`.

Having said that, perhaps in case it could be useful for integation tests later, you Guys may wanna move it to an "all-tests" global `conftest.py` one level above.